### PR TITLE
HPCC-27695 Fix ESP leaks and error reporting

### PR DESCRIPTION
**Pull request by asselitx**
Generated ESP CClient method implementations leak when there are exceptions
thrown in their call to post(). Fix by reference counting the response
object.

SSL plugin is loaded each time the HttpClientContext is created, but not
managed. Instead, load once into an Owned pointer in the module and ensure
it is destroyed on module exit.

Update error messages to have useful information when SOAP client doesn't get
a response from the server.

These issues are seen in the context of the esdl tool calling to the
WsEsdlConfig service.

Signed-off-by: Terrence Asselin <terrence.asselin@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [x] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Cloud-compatibility
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Manual testing with valgrind to confirm fix.

### CODE REVIEW

File: esp/bindings/http/client/httpclient.cpp [hunk 1]
```diff
@@ -951,9 +956,11 @@ HttpClientErrCode CHttpClient::postRequest(ISoapMessage &req, ISoapMessage& resp
     int statusCode = atoi(status.str());
     char statusClass = '0';
     if(status.length() > 0)
+    {
         statusClass = status.charAt(0);
+        errmsg.clear().appendf("HTTP Status %s", status.str());
```
Comment (afishbeck) [inline esp/bindings/http/client/httpclient.cpp:961]: Very minor, but if you don't move the errmsg.clear() above the IF you could repeat an error log that was already logged inside the connect() above.

**Review by afishbeck — COMMENTED** [review]
Very minor comment, otherwise I think it looks good.

@ghalliday may want to take a quick look since this is called quite a few places.

**Review by ghalliday — APPROVED** [review]
Looks good - but tony's comment about clear() is still relevant.

**Comment by asselitx** [comment]
Addressed Tony's comment, squashed and changed base branch to candidate-8.8.x

**Review by afishbeck — APPROVED** [review]

**Pull request merged**